staging/src/k8s.io/component-helpers/scheduling/corev1/helpers.go

```diff
@@ -43,3 +47,51 @@ func MatchNodeSelectorTerms(
 	}
 	return nodeaffinity.NewLazyErrorNodeSelector(nodeSelector).Match(node)
 }
+
+// IsExtendedResourceName returns true if:
+// 1. the resource name is not native
+// 2. resource name does not have "requests." prefix,
+// to avoid confusion with the convention in quota
+// 3. it satisfies the rules in IsQualifiedName() after converted into quota resource name
+func IsExtendedResourceName(name v1.ResourceName) bool {
```

Comment:
can you put these functions in their own file `resource_names.go`?

Reply:
Done